eland/operations.py

```diff
@@ -1198,32 +1197,23 @@ def describe(self, query_compiler: "QueryCompiler") -> pd.DataFrame:
 
     def to_pandas(
         self, query_compiler: "QueryCompiler", show_progress: bool = False
-    ) -> None:
-
-        collector = PandasDataFrameCollector(show_progress)
-
-        self._es_results(query_compiler, collector)
+    ) -> pd.DataFrame:
+        df = self._es_results(query_compiler, show_progress)
 
-        return collector._df
+        return df
 
     def to_csv(
         self,
         query_compiler: "QueryCompiler",
         show_progress: bool = False,
         **kwargs: Union[bool, str],
     ) -> None:
-
-        collector = PandasToCSVCollector(show_progress, **kwargs)
-
-        self._es_results(query_compiler, collector)
-
-        return collector._ret
+        df = self._es_results(query_compiler, show_progress)
+        df.to_csv(**kwargs)
```

Comment:
`return df.to_csv()`? We should add a type and test for this.

Reply:
So, we actually won't return this, the path of the file is passed in kwargs, where the csv has to be ingested to. 

We already have good tests in `tests\dataframe\test_to_csv_pytest.py`
What do you think ?

Reply:
From reading [`pandas.DataFrame.to_csv()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html) it looks like if `path_of_buf` is `None` then the CSV is returned as a string. We should probably do and test that as well.